sdk/appconfiguration/src/feature_manager/feature_type.rs

```diff
@@ -0,0 +1,39 @@
+use serde::{Deserialize, Serialize};
+
+use super::{feature_filter::FeatureContext, models::FeaturesFilter};
+
+const FEATURE_TARGETING: &str = "Microsoft.Targeting";
+const FEATURE_PERCENTAGE: &str = "Microsoft.Percentage";
+const FEATURE_TIME_WINDOW: &str = "Microsoft.TimeWindow";
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum Feature {
+    Percentage(FeatureContext),
+    Targeting(FeatureContext),
+    TimeWindow(FeatureContext),
+    OnOff(bool),
```

Comment:
I'm having trouble finding this in the service documentation or in other SDKs.  Is there reference documentation for this feature somewhere?

Reply:
There two options for using such feature type : 

1.  In feature manager, has possibility to create feature toggle without any  `client_filters`, like simple boolean feature (on/off => "enabled": true|false)
```{
	"id": "on_off",
	"description": "",
	"enabled": true,
	"conditions": {
		"client_filters": []
	}
}```
2. I assume it would be great options to developers, has possibility to override remote value of any feature toggles types. So this OnOff stand for this, you could check how it used in example  dev_on_off_feature_manager

Reply:
I recognize these are potentially useful, but this is opinionated functionality that is unlike any of the other SDKs.

Consider [Microsoft.FeatureManagement](https://github.com/microsoft/FeatureManagement-Dotnet), a dotnet library that builds on top of the Azure SDK for Net, and implements ideas such as what you're proposing.

@heaths do you have thoughts on this?

Reply:
I'm afraid I don't really have much context on App Configuration, but In .NET, we expose a `string Value` property from the [base class](https://github.com/Azure/azure-sdk-for-net/blob/b2a2f3e281f56fb74298f7e975cef7fa8e3ec830/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/ConfigurationSetting.cs#L65) and, in the derived class, effectively make sure we can convert as needed (even for complex objects via JSON) or throw otherwise): https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/FeatureFlagConfigurationSetting.cs

That said, newer libraries use a slightly different approach since .NET nor C# has unions / enums like Rust e.g.,

* [DocumentField](https://github.com/Azure/azure-sdk-for-net/blob/b2a2f3e281f56fb74298f7e975cef7fa8e3ec830/sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentField.cs#L97)
* [DocumentFieldValue](https://github.com/Azure/azure-sdk-for-net/blob/b2a2f3e281f56fb74298f7e975cef7fa8e3ec830/sdk/formrecognizer/Azure.AI.FormRecognizer/src/DocumentFieldValue.cs#L14)

This is an evolving pattern and has been used in a couple newer SDKs or features in older SDKs, like Key Vault.

That said, because Rust has enums like it does, a similar approach such as this makes more sense to me. What do you think, @tg-msft, @JeffreyRichter, @johanste, @bterlson? (Rust enums are effectively like unions with more context - TypeScript has similar enums.)

Reply:
Shouldn't this boil down to being an "extensible enum". We never discussed the best way to expose these in Rust; we should have that conversation and then I think this would be a usage of it.